DQ_test.py

```diff
@@ -24,8 +24,27 @@
 import numpy
 from dqrobotics import *
 from DQ_test_facilities import get_list_of_dq_from_mat
+from DQ_test_facilities import get_list_of_matrices_from_mat
 
-mat = scipy.io.loadmat('DQ_test.mat')
+"""
+Note from Juan Jose Quiroz Omana:
+
+I generated new data (DQ_test.mat) using DQ_test.m to include Q8 operations. 
+The legacy data (October 10, 2019) is in DQ_test_october_10_2019.mat.
+However, The test_ad fails with 'DQ_test.mat' with the following assertion:
+
+AssertionError: 
+Arrays are not almost equal to 12 decimals
+Error in Ad
+Mismatched elements: 1 / 8 (12.5%)
+Max absolute difference: 1.8189894e-12
+Max relative difference: 5.05591348e-16
+
+This is the reason to use 'DQ_test_october_10_2019.mat' for all tests except 
+test_of_Q4 and test_of_Q8, which use 'DQ_test.mat.
+"""
+mat = scipy.io.loadmat('DQ_test_october_10_2019.mat')
```

Comment:
@juanjqo

1.  I suppose that eventually we'll need to parse the name of these `.mat` files, so we can make them a little more computer friendly. For instance, rename to `DQ_test_DD_MM_YYYY.mat`.

2. In the local variables, instead of `mat` it will probably be easier to understand if you change it to `mat_10_10_2019`.

3. (needs further discussion @dqrobotics/developers ) It is strange that we are relying on two different mat files simultaneously, I strongly believe we need further discussion. My current understanding is that if we hope that the dual quaternion multiplication is within a threshold of 12 decimals (i.e. `DQ_threshold `), it does not seem particularly reasonable to expect that the multiplication among three dual quaternions will be within the same threshold.

Support information for 3.
- The [`assertEqual`](https://github.com/dqrobotics/python-tests/blob/4cd0f1ee9989c2ffb91a55bd743ecdfdb3751eaa/DQ_test.py#L87) used for testing multiplication does not have an explicit threshold but it uses the [DQ equality](https://github.com/dqrobotics/cpp/blob/93d4a05eb7ba1ab7c903f0710b76e2700d6a5fcc/src/DQ.cpp#L1302), which, itself, uses a elementwise `DQ_threshold.`

Reply:
@dqrobotics/developers For our information (FOI?):

https://www.mathworks.com/matlabcentral/answers/19736-identical-math-operations-in-c-and-matlab-differ-by-1-10-11-in-this-case

Reply:
@dqrobotics/developers I'll leave this last topic without `Resolve conversation` so that we can come back to it in due time.

For now, as @juanjqo mentioned, we'll use a relaxed threshold for `Ad` and `Adsharp`, so that we can move on with this and other improvements in the library.

We can come back to this with a more detail analysis and better solution in due course.